Empty custom config enables rules disabled by default

## Expected Behavior

When I create an empty custom config file and enable it though `detekt { config ... }`, it should work as if the custom config wasn't enabled.

## Observed Behavior
My Gradle config is as follows:

```
detekt {
    buildUponDefaultConfig = true
    config = files("$projectDir/detekt-custom-config.yml")
}
```

and the config file is empty. When I run Detekt, I'm getting

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':backend:detekt'.
> Rule 'AbsentOrWrongFileLicense': License template file not found at `/home/piotr/repos/teams/backend/license.template`.
  Create file license header file or check your running path.
```

I customized one rule that I needed and it now seems to work correctly.

## Steps to Reproduce

See above.

## Context

This issue caused confusion as the behavior is counter-intuitive, or if it's somehow by design, it's not documented well enough.

## Your Environment

* Version of detekt used: 1.15.0
* Version of Gradle used (if applicable): 6.6.1
* Operating System and version: Ubuntu 20.04
* Link to your project (if it's a public repository): (private repo - easy to reproduce given above description)

--- a/detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt
+++ b/detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/Config.kt
@@ -50,7 +50,7 @@ interface Config {
         /**
          * An empty configuration with no properties.
          * This config should only be used in test cases.
-         * Always returns the default value except when 'active' is queried, it returns true .
+         * Always returns the default value except when 'active' is queried, it returns true.
          */
         val empty: Config = EmptyConfig
 

--- a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/config/YamlConfig.kt
+++ b/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/config/YamlConfig.kt
@@ -72,7 +72,7 @@ class YamlConfig internal constructor(
                 Yaml().loadAs(it, Map::class.java) as Map<*, *>?
             }.getOrElse { throw Config.InvalidConfigurationError(it) }
             if (map == null) {
-                Config.empty
+                YamlConfig(emptyMap())
             } else {
                 YamlConfig(map as Map<String, Any>)
             }

--- a/detekt-core/src/main/resources/default-detekt-config.yml
+++ b/detekt-core/src/main/resources/default-detekt-config.yml
@@ -17,11 +17,18 @@ processors:
   active: true
   exclude:
     - 'DetektProgressListener'
+  # - 'KtFileCountProcessor'
+  # - 'PackageCountProcessor'
+  # - 'ClassCountProcessor'
   # - 'FunctionCountProcessor'
   # - 'PropertyCountProcessor'
-  # - 'ClassCountProcessor'
-  # - 'PackageCountProcessor'
-  # - 'KtFileCountProcessor'
+  # - 'ProjectComplexityProcessor'
+  # - 'ProjectCognitiveComplexityProcessor'
+  # - 'ProjectLLOCProcessor'
+  # - 'ProjectCLOCProcessor'
+  # - 'ProjectLOCProcessor'
+  # - 'ProjectSLOCProcessor'
+  # - 'LicenseHeaderLoaderExtension'
 
 console-reports:
   active: true

--- a/detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/printer/rulesetpage/ConfigPrinter.kt
+++ b/detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/printer/rulesetpage/ConfigPrinter.kt
@@ -94,11 +94,18 @@ object ConfigPrinter : DocumentationPrinter<List<RuleSetPage>> {
         active: true
         exclude:
           - 'DetektProgressListener'
+        # - 'KtFileCountProcessor'
+        # - 'PackageCountProcessor'
+        # - 'ClassCountProcessor'
         # - 'FunctionCountProcessor'
         # - 'PropertyCountProcessor'
-        # - 'ClassCountProcessor'
-        # - 'PackageCountProcessor'
-        # - 'KtFileCountProcessor'
+        # - 'ProjectComplexityProcessor'
+        # - 'ProjectCognitiveComplexityProcessor'
+        # - 'ProjectLLOCProcessor'
+        # - 'ProjectCLOCProcessor'
+        # - 'ProjectLOCProcessor'
+        # - 'ProjectSLOCProcessor'
+        # - 'LicenseHeaderLoaderExtension'
     """.trimIndent()
 
     private fun defaultConsoleReportsConfiguration(): String = """

--- a/docs/pages/configurations.md
+++ b/docs/pages/configurations.md
@@ -140,11 +140,18 @@ processors:
     active: true
     exclude:
         - 'DetektProgressListener'
+        # - 'KtFileCountProcessor'
+        # - 'PackageCountProcessor'
+        # - 'ClassCountProcessor'
         # - 'FunctionCountProcessor'
         # - 'PropertyCountProcessor'
-        # - 'ClassCountProcessor'
-        # - 'PackageCountProcessor'
-        # - 'KtFileCountProcessor'
+        # - 'ProjectComplexityProcessor'
+        # - 'ProjectCognitiveComplexityProcessor'
+        # - 'ProjectLLOCProcessor'
+        # - 'ProjectCLOCProcessor'
+        # - 'ProjectLOCProcessor'
+        # - 'ProjectSLOCProcessor'
+        # - 'LicenseHeaderLoaderExtension'
 ```
 
 ## Config JSON Schema

